packages/react-instantsearch-hooks-dom/src/widgets/__tests__/InfiniteHits.test.tsx

```diff
@@ -13,63 +13,106 @@ type CustomHit = {
   somethingSpecial: string;
 };
 
-const searchClient = createSearchClient({
-  search: (requests) =>
-    Promise.resolve(
-      createMultiSearchResponse(
-        ...requests.map((request) => {
-          const { hitsPerPage = 3, page = 0 } = request.params!;
-          const hits = Array.from({ length: hitsPerPage }, (_, i) => {
-            const offset = hitsPerPage * page;
-            return {
-              objectID: (i + offset).toString(),
-              somethingSpecial: String.fromCharCode(
-                'a'.charCodeAt(0) + i + offset
-              ),
-            };
-          });
-
-          return createSingleSearchResponse<CustomHit>({
-            hits,
-            page,
-            nbPages: 10,
-            hitsPerPage,
-            index: request.indexName,
-          });
-        })
-      )
-    ),
-});
-
 describe('InfiniteHits', () => {
+  let searchClient: ReturnType<typeof createSearchClient>;
+  beforeEach(() => {
+    searchClient = createSearchClient({
+      search: jest.fn((requests) =>
+        Promise.resolve(
+          createMultiSearchResponse(
+            ...requests.map((request) => {
+              const { hitsPerPage = 3, page = 0 } = request.params!;
+              const hits = Array.from({ length: hitsPerPage }, (_, i) => {
+                const offset = hitsPerPage * page;
+                return {
+                  objectID: (i + offset).toString(),
+                  somethingSpecial: String.fromCharCode(
+                    'a'.charCodeAt(0) + i + offset
+                  ),
+                };
+              });
+
+              return createSingleSearchResponse<CustomHit>({
+                hits,
+                page,
+                nbPages: 10,
+                hitsPerPage,
+                index: request.indexName,
+              });
+            })
+          )
+        )
+      ),
+    });
+  });
```

Comment:
Why do you need a fresh client for each test? Could we clear the mocks instead or is there something else?